spec/models/provider_relationship_permissions_spec.rb

```diff
@@ -146,4 +146,55 @@
       expect(relationship.permit?(:view_diversity_information, training_provider)).to be true
     end
   end
+
+  describe "#governs_open_course?" do
+    let(:training_provider) { create(:provider) }
+    let(:ratifying_provider) { create(:provider) }
+    let!(:relationship) do
+      create(
+        :provider_relationship_permissions,
+        training_provider: training_provider,
+        ratifying_provider: ratifying_provider,
+        )
+    end
+
+    context "there are no courses" do
+      it "returns false" do
+        expect(relationship.governs_open_course?).to be(false)
+      end
+    end
+
+    context "there is an unopened course" do
+      let!(:course) { create(:course, provider: training_provider, accredited_provider: ratifying_provider) }
+
+      it "returns false" do
+        expect(relationship.governs_open_course?).to be(false)
+      end
+    end
+
+    context "there is an open course" do
+      let!(:course) { create(:course, :open_on_apply, provider: training_provider, accredited_provider: ratifying_provider)}
+
+      it "returns false" do
+        expect(relationship.governs_open_course?).to be(true)
+      end
+    end
+  end
+
+  describe ".govern_open_course" do
+    def create_relationship_with_course(traits: [])
```

Comment:
Minor, but this might be clearer
```suggestion
    def create_relationship_with_course(course_traits: [])
```